### PR TITLE
retry apt when locked

### LINKED ISSUE #952: OCI build failures due to apt lock
**What steps did you take and what happened:**
When building a Ubuntu 20.04 OCI, the build very often fails in the initial `dist-upgrade` task due to apt lock being held by another process. I think this due to some of the apt related modules in the cloud-config & cloud-final that comes in from the stock oci image. 

```
    oracle-oci: TASK [setup : perform a dist-upgrade] ******************************************
    oracle-oci: fatal: [default]: FAILED! => {"changed": false, "msg": "'/usr/bin/apt-get dist-upgrade ' failed: E: Could not get lock /var/lib/dpkg/lock-frontend. It is held by process 2228 (apt)\nE: Unable to acquire the dpkg frontend lock (/var/lib/dpkg/lock-frontend), is another process using it?\n", "rc": 100, "stdout": "", "stdout_lines": []}
    oracle-oci:
```

Ansible 2.13.x has the support for a `lock_timeout` option in the apt module but did not seem to help as it would fail with the same reason. So, the solution I'm thinking is to check the result of the task output and retry a few times. 

**What did you expect to happen:**
Build successful

**Anything else you would like to add:**

**Environment:**

Project ([Image Builder for Cluster API](https://github.com/kubernetes-sigs/image-builder/tree/master/images/capi), [kube-deploy/imagebuilder](https://github.com/kubernetes-sigs/image-builder/tree/master/images/kube-deploy/imagebuilder), [konfigadm](https://github.com/kubernetes-sigs/image-builder/tree/master/images/konfigadm)):

Additional info for Image Builder for Cluster API related issues:

- OS (e.g. from `/etc/os-release`, or `cmd /c ver`):
- Packer Version:
- Packer Provider:
- Ansible Version: 
- Cluster-api version (if using):
- Kubernetes version: (use `kubectl version`):

/kind bug
[One or more /area label. See https://github.com/kubernetes-sigs/cluster-api/labels?q=area for the list of labels]

### PR DESCRIPTION
**Pull request by kkeshavamurthy**
What this PR does / why we need it:
Fixes #952 
Which issue(s) this PR fixes (optional, in fixes #<issue number>(, fixes #<issue_number>, ...) format, will close the issue(s) when PR gets merged): Fixes #

**Additional context**
Add any other context for the reviewers

### CODE REVIEW

**Comment by kkeshavamurthy** [comment]
CC: @shyamradhakrishnan

File: images/capi/ansible/roles/setup/tasks/debian.yml [hunk 1]
```diff
@@ -56,6 +56,10 @@
     force_apt_get: True
     update_cache: True
     upgrade: dist
+  register: apt_lock_status
+  until: apt_lock_status is not failed
+  retries: 5
+  delay: 30
```
Comment (jsturtevant) [inline images/capi/ansible/roles/setup/tasks/debian.yml:62]: 30 seconds is pretty long delay.  How did you come to that number?

Comment (kkeshavamurthy) [reply images/capi/ansible/roles/setup/tasks/debian.yml:62]: In testing I had 10s and that worked without any issues. I'll revert it back to 10 then.

Comment (jsturtevant) [reply images/capi/ansible/roles/setup/tasks/debian.yml]: OK. Not opposed just curious.

Comment (kkeshavamurthy) [reply images/capi/ansible/roles/setup/tasks/debian.yml]: Yeah 30s did seem a bit long after you mentioned it. 10s should do the job :)

**Pull request merged**